src/shmem_synchronization.h

```diff
@@ -141,17 +141,37 @@ shmem_internal_fence(void)
     } while(0)
 #elif defined(USE_OFI)
 
+#if defined(ENABLE_HARD_POLLING) || defined(USE_ON_NODE_COMMS)
```

Comment:
+1, this passage is redundant.  The macro re-definition should have generated a compiler warning.